Sources/MapboxNavigation/NavigationViewController.swift

```diff
@@ -285,6 +305,8 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
     
     var viewObservers: [NavigationComponentDelegate] = []
     
+    var mapTileStore: TileStore?
```

Comment:
Had to store it here to be able to reach it during `loadView` and configure `NavigationView`. Tile storage is configured at initialisation time for `MapView` which forces piping this as init parameter through `NavigationView` and `NavigationMapView`. Pretty much the same happens for Navigator.
Apart of ravaging the init API, it also makes tile store setup very sensitive to init order. If any component is created before `NavigationViewController.init` - it will ignore `NavigationOptions` setup.

Reply:
> If any component is created before `NavigationViewController.init` - it will ignore `NavigationOptions` setup.

This suggests that maybe those components should also accept a NavigationOptions in their initializers, or that the tile store should be independent of NavigationOptions.

Reply:
>that the tile store should be independent of NavigationOptions

Not sure to follow this part. Did you mean to extract TileStore configuration from `NavigationOptions`? But that ruins concept of `NavigationOptions` as single customization source.
Having components accept `NavigationOptions` seems and overkill since it just needs a TileStore bit. But it brings us back to having `TileStoreLocation` as a standalone type suitable for configuring components. I guess I'll try this case